cumulusci/robotframework/utils.py

```diff
@@ -199,3 +200,35 @@ def run_with_retry(self, *args, **kwargs):
     else:
         # Decorator was used without arguments
         return decorate(target)
+
+
+class PerfJSONConverter:
+    def __init__(self, jsondata):
+        self.data = json.loads(jsondata)
+
+    def to_dict(self, include_raw=False):
+        rc = {
+            metric["metrics"] + "-" + metricType: metric[metricType]
+            for metricType in ("totalTime", "totalCalls")
+            for metric in self.data["summary"]
+        }
+        if include_raw:
+            rc["_raw"] = self.data
+        return rc
+
+    def to_csv(self):
+        """Helper function to generate CSV-like data from performance metrics JSON"""
+
+        def row(values):
+            return ",".join(values) + "\n"
```

Comment:
is there a danger that `values` itself may have newlines or commas in it? This is where using a csv library would help.

Reply:
No, its just a list of numbers.

And the sense in which it is CSV-like is that it has a header line in Python comment syntax (because there is no CSV comment syntax) for human readability.

That's what this line does:

        res += "#" + row(metrics[0].keys())

If you strip off that line the rest of the lines can be interpreted as CSV safely.

Reply:
It's been years since I've done much with csv files, but IIRC it's a fairly well established convention that the first row can (should?) be column names. With or without the leading `#`, the line should probably be able to be parsed as csv so that automated tools won't crash and burn trying to parse the first row. 

I'm not going to get overly passionate about this, but I think separating the keys by commas would be worth doing. If there's a chance some of the keys have commas in them, they would need to be properly escaped.